### PR TITLE
Feature/tests

### PR DESCRIPTION
**Pull request by garethnunns**
Base test suite using [Mocha](https://mochajs.org/) & [Assert](https://nodejs.org/api/assert.html).

Checks the indexes exist and have valid titles/links #79

Makes sure all links are valid as mentioned in #72 

Currently brings up errors of [missing images](https://manual.avolites.com/docs/controlling-fixtures/using-the-select-buttons-and-wheels/#shutter-blades-keystone), maybe fix those in another PR...

### CODE REVIEW

**Comment by congoblue** [comment]
The "Blades" image flagged as missing is a new image for v14 and is present in the v13+ document

**Comment by garethnunns** [comment]
> The "Blades" image flagged as missing is a new image for v14 and is present in the v13+ document

I don't think it's been committed into this [master](https://github.com/AvolitesLtd/TitanManual/tree/master/website/static/docs/images/) as it's not present on the PDF/website: https://manual.avolites.com/docs/controlling-fixtures/using-the-select-buttons-and-wheels/#shutter-blades-keystone

[There are a few other errors too](https://github.com/AvolitesLtd/TitanManual/pull/86/checks?check_run_id=1007184044#step:5:7602):

````
  6808 passing (554ms)
  19 failing

  1) Docs
       next - /home/runner/work/TitanManual/TitanManual/docs/patching/changing-the-patch.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Exchange-Mapping.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Exchange-Mapping.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/docs/patching/changing-the-patch.md" (with the alt text "Exchange Mapping")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  2) Docs
       13.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/chases/chase-options.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Playback-Options-playback-tab-for-a-chase.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Playback-Options-playback-tab-for-a-chase.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/chases/chase-options.md" (with the alt text "Playback Options (playback tab) for a chase")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  3) Docs
       13.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Mix.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Mix.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Colour Mix")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  4) Docs
       13.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Picker.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Picker.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Colour Picker")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  5) Docs
       13.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Channels.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Channels.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Colour Channels")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  6) Docs
       13.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Filters.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Filters.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Colour Filters")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  7) Docs
       13.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Blades.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Blades.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Blades")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  8) Docs
       13.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/cues/playback-options.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Playback-Options-window-for-cue-showing-playback-tab.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Playback-Options-window-for-cue-showing-playback-tab.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/cues/playback-options.md" (with the alt text "Playback Options window for cue showing playback tab")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  9) Docs
       13.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/patching/changing-the-patch.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Exchange-Mapping.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Exchange-Mapping.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/patching/changing-the-patch.md" (with the alt text "Exchange Mapping")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  10) Docs
       13.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/titan-basics/workspace-windows.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Appearance-Menu.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Appearance-Menu.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-13.0/titan-basics/workspace-windows.md" (with the alt text "Appearance Menu")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  11) Docs
       12.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/chases/chase-options.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Playback-Options-playback-tab-for-a-chase.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Playback-Options-playback-tab-for-a-chase.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/chases/chase-options.md" (with the alt text "Playback Options (playback tab) for a chase")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  12) Docs
       12.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Mix.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Mix.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Colour Mix")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  13) Docs
       12.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Picker.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Picker.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Colour Picker")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  14) Docs
       12.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Channels.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Channels.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Colour Channels")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  15) Docs
       12.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Filters.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Colour-Filters.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Colour Filters")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  16) Docs
       12.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Blades.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Blades.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/controlling-fixtures/using-the-select-buttons-and-wheels.md" (with the alt text "Blades")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  17) Docs
       12.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/cues/playback-options.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Playback-Options-window-for-cue-showing-playback-tab.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Playback-Options-window-for-cue-showing-playback-tab.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/cues/playback-options.md" (with the alt text "Playback Options window for cue showing playback tab")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  18) Docs
       12.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/patching/changing-the-patch.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Exchange-Mapping.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Exchange-Mapping.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/patching/changing-the-patch.md" (with the alt text "Exchange Mapping")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)

  19) Docs
       12.0 - /home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/titan-basics/workspace-windows.md
         Local images
           Image "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Appearance-Menu.png":

      AssertionError [ERR_ASSERTION]: Could not find "/home/runner/work/TitanManual/TitanManual/website/static/docs/images/Appearance-Menu.png"
                  Linked from "/home/runner/work/TitanManual/TitanManual/website/versioned_docs/version-12.0/titan-basics/workspace-windows.md" (with the alt text "Appearance Menu")
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (/home/runner/work/TitanManual/TitanManual/test/test.docs.js:77:17)
      at processImmediate (internal/timers.js:456:21)
````

File: parse/search.js [hunk 1]
```diff
@@ -3,17 +3,54 @@ const path = require('path');
 const fs = require('fs');
 const avoParse = require('./avoParse')
 
-avoParse.getVersions().forEach(function(version) {
```
Comment (garethnunns) [inline parse/search.js]: Might be worth using these library functions

Comment (Farrser) [reply parse/search.js]: Ah yeah my bad, will get it changed back

**Review by Farrser — APPROVED** [review]

**Pull request merged**